packages/types/lib/IStatistic.ts

```diff
@@ -0,0 +1,101 @@
+// eslint-disable-next-line
+import type { EventEmitter } from 'events';
+import type { ILink } from '@comunica/bus-rdf-resolve-hypermedia-links';
+import type { ActionContextKey } from '@comunica/core';
+import type { IActionContext } from './IActionContext';
+import type { IQuerySource } from './IQuerySource';
+
+export interface IStatistic {
+  /**
+   * Query string of tracked execution
+   */
+  query: string, 
+
+  statisticEvents: EventEmitter;
+
+  getEmitter: () => EventEmitter;
```

Comment:
I did a quick search here, and it looks like you're only emitting on-data events.
So for type-safety, I would propose just adding the following two methods instead:
```
addListener(cb: () => void): void;
emit(data: T): void;
```
For the `T`, you'll have to add a generic type to the `IStatistic` to indicate the type of data that is emitted.

Reply:
Right, that does make more sense.